### PR TITLE
End of ClothKiller

### LINKED ISSUE #2380: ClothKiller Cleanup
# Description
Creating this for myself to track the stuff that was waiting on the reorg. There's several changes to be made.

- [x] Add AssetUsageDetector plugin to simplify finding usages of assets
- [x] Move ClothKiller / Construction into the correct place in the new resources folder structure
- [x] Migrate the old ItemAttributes component to the new ItemAttributesV2 and delete the IITemAttributes interface and the old component.
- [x] Migrate the old Clothing component to the new ClothingV2 and delete the old one.
- [x] Rename NamedSlots.exosuit to outerwear to better match their actual purpose
- [x] Remove all Spawnable assets, ClothPopulator, PrefabPopulator, and Custom Slot Populators. All slot populators will simply use prefab variants instead. Remove all fields that reference Slot Populators.
- [x] Remove all the extra fields from all of the cloth data assets that aren't needed anymore. Consider refactoring BeltData to use one of the other cloth data assets since I don't know if it has any specific logic for it anymore (maybe it can just use ContainerData).
- [x] Delete ResourceTracker stuff that isn't needed anymore
- [x] Delete all the lifecycle logic related to cloths, since it uses prefab variants now.
- [x] Delete all the cloth data caches / dictionaries / managers
- [x] Remove cloth-specific logic from dev spawner
- [x] Delete all UniItem prefabs - UniBackPack, UniCloth, UniHeadset, UniFloorTile

### PR DESCRIPTION
**Pull request by chairbender**
### Purpose
Finally finishes all the cleanup needed after clothkiller PR.

Fixes #2380 

No more slot populators, updating Wiki on that. They just reference prefabs now, much simpler to setup. If you need custom slot population logic you can build it into the prefab itself so it inits itself on spawn (see AutoInitID.prefab for example).

Fixes a bug where ID cards had wrong icon in inventory on spawn.

Checked in a bunch of prefabs that were showing as modified for me so hopefully others don't have to. None of the prefabs except AutoInitID.prefab had any actual changes, so if there's a conflict it can be overwritten.

Exosuit namedslot is now outerwear, which more accurately reflects what it is used for. I renamed the Suitwear folder / base prefab variant to Uniform to match the namedslot, since I think uniform is actually the correct name for it.

No more spawnable assets, only implementors of the ISpawnable interface to be referenced in code. Removes the tedium of referencing things to spawn, you can just select prefabs directly rather than wrapping them in a spawnable asset.

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  Any new files are named using PascalCase (to avoid issues on case sensitive file systems)
- [x]  Any new / changed components follow the [Component Development Checklist](https://github.com/unitystation/unitystation/wiki/Component-Development-Checklist)
- [x]  Any new objects / items follow the [Creating Items and Objects Guide](https://github.com/unitystation/unitystation/wiki/Creating-Items-and-Objects%3A-Now-With-Prefab-Variants) (especially concerning the use of prefab variants)
- [x]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)
- [ ]  This PR has been tested with round restarts.
- [x]  This PR has been tested on moving / rotating / rotated-before-joining matrices (if applicable)

### Notes:
_Please enter any other relevant information here_

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/IDCard.cs [hunk 1]
```diff
@@ -76,19 +84,54 @@ public override void OnStartClient()
 		base.OnStartClient();
 	}
 
+	public void OnSpawnServer(SpawnInfo info)
+	{
+		hasAutoInit = false;
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Items/IDCard.cs:89]: how does that `hasAutoInit` work?

**Comment by fomalsd** [comment]
Getting  following errors/warnings:
```
'SpawnablePrefab' is missing the class attribute 'ExtensionOfNativeClass'!
UnityEngine.Resources:LoadAll(String)
CustomNetworkManager:SetSpawnableList() (at Assets/Scripts/Managers/NetworkManagement/CustomNetworkManager.cs:50)
CustomNetworkManager:Start() (at Assets/Scripts/Managers/NetworkManagement/CustomNetworkManager.cs:37)
```
```
The class named 'SpawnablePrefab' is not derived from MonoBehaviour or ScriptableObject!
UnityEngine.Resources:LoadAll(String)
CustomNetworkManager:SetSpawnableList() (at Assets/Scripts/Managers/NetworkManagement/CustomNetworkManager.cs:50)
CustomNetworkManager:Start() (at Assets/Scripts/Managers/NetworkManagement/CustomNetworkManager.cs:37)
```
Shouldn't SpawnablePrefab be a ScriptableObject? At the moment it's just a pure class.

Otherwise it's looking good!

**Comment by chairbender** [comment]
No, as mentioned in PR spawnables as assets are removed, they create needless tedium and complexity compared with just directly referencing prefabs. The spawnables interfaces are only used internally now.

The error is just unity complaining about old reference to old spawnableprefab asset, even though I have removed all fields of that type from all components I think the config still is there in the yaml. Can fix it.

**Comment by chairbender** [comment]
should be fixed now

**Review by fomalsd — APPROVED** [review]

**Comment by fomalsd** [comment]
all fine, thanks!

**Pull request merged**